src/common.jl

```diff
@@ -24,11 +24,11 @@ const RealFP = Union{Float32, Float64}
 
 fptype(::Type{T}) where {T<:Union{Float32,Bool,Int8,UInt8,Int16,UInt16}} = Float32
 fptype(::Type{T}) where {T<:Union{Float64,Int32,UInt32,Int64,UInt64,Int128,UInt128}} = Float64
-fptype(::Type{Complex64}) = Complex64
-fptype(::Type{Complex128}) = Complex128
+fptype(::Type{Compat.ComplexF32}) = Compat.ComplexF32
```

Comment:
The `Compat.` prefix isn't needed for these (or `Nothing` below), since they're exported from Compat. Also, just personal preference, but I think it might be clearer to spell out the `Complex` type explicitly, e.g. `::Type{Complex{Float32}}`, which will work across versions.

Reply:
It might be simpler to define 
```
fptype(::Type{T}) where T = float(T)
fptype(::Type{T}) where {T<:Union{Bool,Int8,UInt8,Int16,UInt16}} = Float32
```